### PR TITLE
Upgrade nippy to 2.13.0

### PR DESCRIPTION
**Pull request by Skiggz**
* The motivation here is actually to update the transient "com.taoensso/encore" dependency to a much newer version
* nippy 2.10.0 depends on a version of encore that dates back 2 years ago, 71 releases old x.x
* We had made this change before, and errors were being thrown, now we are making this change again and catching those
* When failing to deserialize old values, we will simply fail silently and assume the cache will be reloaded with a compatible serialized value
* Eventually (or never?) we can remove this but it shouldn't be detrimental in any case

@taubeklavs @skazhy @zarlen 

PS. I'm wondering if this should be a flag as well, that defaults to the try, but can avoid it if the client knows it's only got recently frozen items in cache (or cleared the cache).

### CODE REVIEW

**Comment by skazhy** [comment]
You can explicitly exclude nippy when adding full.cache as dependency & then adding nippy version which you need. This is a really high risk change, since there are lot of implicit dependencies on this.

File: test/full/t_cache.clj [hunk 1]
```diff
@@ -3,12 +3,35 @@
             [full.core.sugar :refer :all]
             [full.async :refer :all]
             [full.cache :as cache]
-            [clojure.core.async :as async :refer [<!!]]))
+            [clojure.core.async :as async :refer [<!!]]
+            [taoensso.nippy :as nippy]))
+
+(def ^:private some-bytes
```
Comment (FnTm) [inline test/full/t_cache.clj]: This should really be named better. Is there some specific meaning to these particular bytes? Or are they really just random bunch of bytes?

**Review by skazhy — CHANGES_REQUESTED** [review]
As mentioned before - this change either needs to happen in the service using this, or bump version to `1.1.0`, so we can update cache with old nippy, if needed.

File: src/full/cache.clj [hunk 2]
```diff
@@ -94,6 +94,20 @@
...
+  To fix that and keep moving with the later version of nippy (and more importantly encore)
+  catch deserialization issues and log them, assuming the caller will simply reload the cache
+  with a freshly reserialized value that should successfully deserialize next time."
+  [k val]
+  (try
+    (nippy/thaw val)
+    (catch Throwable e
+      (do
+        ;; Specifically do not log stack trace since that is slow and noisy and we know it was likely bad bytes
+        (log/debug "Failed to thaw value in cache for " k ", should be refreshed. Error: " (.getMessage e))
```
Comment (skazhy) [inline src/full/cache.clj]: `full.core.log/debug` returns nil, so `do` block is redundant here.

File: test/full/t_cache.clj (hunk 1 above)
Comment (Skiggz) [reply test/full/t_cache.clj]: They are random (well, they spell `FullContact!`) but it was for a test of a random input of bytes nippy wouldn't parse lol.

File: src/full/cache.clj (hunk 2 above)
Comment (Skiggz) [reply src/full/cache.clj]: Oh yeah duh.

**Comment by Skiggz** [comment]
Thanks for the feedback peeps.

I simplified this via karlis' suggestion. Now it's just a `2.0.0` version bump and I made deserialization a part of the normal fetch/throw flow that the interface is already built for (i.e. the `:throw?` option). I removed all the other mumbo jumbo so it's a pretty basic change. 

As for _risky_, I think the only risk here is 100% cache miss on day 1 that would quickly decline as the cache resets itself, assuming the cache is being used in normal fashion. If `:throw?` is being specified, then yes, error rates would spike until that flag was unset and/or the cache was cleared. For extra credit, I've added this note to the README.

**Comment by Skiggz** [comment]
@skazhy bump in case you didn't see last comment/changes.

**Comment by skazhy** [comment]
Rebased in master as `630065653b8`.

**Pull request closed without merging**